apps/desktop/src/autofill/services/desktop-fido2-user-interface.service.ts

```diff
@@ -71,36 +88,90 @@ export class DesktopFido2UserInterfaceSession implements Fido2UserInterfaceSessi
     private cipherService: CipherService,
     private accountService: AccountService,
     private logService: LogService,
-    private messagingService: MessagingService,
     private router: Router,
     private desktopSettingsService: DesktopSettingsService,
+    private windowObject: NativeWindowObject,
   ) {}
 
   private confirmCredentialSubject = new Subject<boolean>();
   private createdCipher: Cipher;
 
+  private availableCipherIdsSubject = new BehaviorSubject<string[]>(null);
+  /**
+   * Observable that emits available cipher IDs once they're confirmed by the UI
+   */
+  get availableCipherIds$(): Observable<string[]> {
+    return this.availableCipherIdsSubject.pipe(
+      filter((ids) => ids != null),
+      take(1),
+    );
+  }
+
+  private chosenCipherSubject = new Subject<string>();
+
   // Method implementation
-  async pickCredential(
-    params: PickCredentialParams,
-  ): Promise<{ cipherId: string; userVerified: boolean }> {
-    this.logService.warning("pickCredential desktop function", params);
+  async pickCredential({
+    cipherIds,
+    userVerification,
+    assumeUserPresence,
+    masterPasswordRepromptRequired,
+  }: PickCredentialParams): Promise<{ cipherId: string; userVerified: boolean }> {
+    this.logService.warning("pickCredential desktop function", {
+      cipherIds,
+      userVerification,
+      assumeUserPresence,
+      masterPasswordRepromptRequired,
+    });
 
     try {
-      await this.showUi();
+      // Check if we can return the credential without user interaction
+      // TODO: Assume user presence is undefined
+      if (cipherIds.length === 1 && !masterPasswordRepromptRequired) {
```

Comment:
question/issue: why? Why not use `assumeUserPresence`?
```suggestion
      // Check if we can return the credential without user interaction
      if (assumeUserPresence && cipherIds.length === 1 && !masterPasswordRepromptRequired) {
```

Reply:
Need to look into this more, but `assumeUserPresence` is undefined so the condition will always be false. Might be because it needs to be added in some call somewhere though?

Reply:
Seems to be missing from `desktop-autofill.service.ts`

Reply:
Resolved by adding it elsewhere. We should safely assume UP was done by the OS.